File: backend/app/schemas/integration.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.integration import IntegrationType


class IntegrationCreate(BaseModel):
    type: IntegrationType
    name: str
    config: dict          # plaintext — encrypted server-side
    is_active: bool = True
    tenant_id: UUID | None = None  # None = global (super admin only)


class IntegrationUpdate(BaseModel):
    name: str | None = None
    config: dict | None = None
    is_active: bool | None = None


class IntegrationRead(BaseModel):
    id: UUID
    tenant_id: UUID | None
    type: IntegrationType
    name: str
    is_active: bool
    scope: str            # "global" | "tenant"
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TestResult(BaseModel):
    success: bool
    message: str
    latency_ms: float | None = None

File: frontend/src/types/integration.ts
export type IntegrationType = "shodan" | "wazuh" | "openvas" | "nmap" | "zabbix" | "bookstack";

export interface Integration {
  id: string;
  tenant_id: string | null;
  type: IntegrationType;
  name: string;
  is_active: boolean;
  scope: "global" | "tenant";
  created_at: string;
  updated_at: string;
}

export interface TestResult {
  success: boolean;
  message: string;
  latency_ms?: number;
}

export interface IntegrationCreate {
  type: IntegrationType;
  name: string;
  config: Record<string, unknown>;
  is_active?: boolean;
  tenant_id?: string | null;
}

export interface IntegrationUpdate {
  name?: string;
  config?: Record<string, unknown>;
  is_active?: boolean;
}
